### ISSUE TITLE
Can't use install instructions verbatim

### ISSUE THREAD

**Issue by rgregg** [knative/docs#1678]
## Expected Behavior
I should be able to copy and paste the command lines to install Knative, e.g. this topic: https://github.com/knative/docs/blob/master/docs/install/Knative-with-GKE.md

## Actual Behavior
I can't, because we're using the {<version>} macro. I need to update this text to reflect whatever version I'm trying to install. 

## Steps to Reproduce the Problem

1. Try to install Knative from https://github.com/knative/docs/blob/master/docs/install/Knative-with-GKE.md
1. Struggle.

**Comment by duglin** [comment]
I saw that too but assumed it was just a temporary thing because they were making infra changes. If this was meant to be "for real" then +1 to changing it back. There's no need to a template string in there since the docs should be versioned along with the code/releases.

**Comment by rgregg** [comment]
I think the goal was to make it easier to maintain the docs, so that they don't need to be updated each version. However, there's a usability hit to consuming the docs here that I'm not sure we've considered.

**Comment by samodell** [comment]
Yep, we are trying to make maintenance of the docs easier. The docs are versioned per release; we wanted to leverage that by making a variable that would automatically pick up the version from the docs branch, so we wouldn't need to manually update the version number for each release. @duglin  curious if it would be problematic for you to start consuming the docs from knative.dev instead of from Github? The version number will resolve properly on the site, so copy/paste will work perfectly from there.

I've added discussing this change to the Docs WG agenda for Aug. 13th; please drop in if you have strong feelings about this.

**Comment by duglin** [comment]
Picking it up from knative.dev is fine - I'll just need to remember go to there instead of github. What's the lag between what's on github vs knative.dev ? Are they always in-sync?

**Comment by samodell** [comment]
Right now the process is semi-manual. If a PR gets submitted to the website repo, the site is rebuilt, but that happens probably less than once a week on average. For now, @RichieEscarez  has to click a button to rebuild the site with the latest info, and I think he tries to do that pretty regularly. He's also working on automating that process to be tied to PRs in the Docs repo as well as the website repo.

**Comment by rgregg** [comment]
I've mentioned this to @RichieEscarez on another topic as well (importing code files via a macro), but I'm going to mention it again.

I'm strongly of the belief that our GitHub markdown content _must_ be consumable by end users as primary documentation. I see this is a **P0**. Until recently this was achieved, although perhaps unintentionally.

We're currently moving in the direction of optimizing for engineering effort at the expense of consumable documentation source. I know this is how our system works inside Google, and it makes our the doc source nearly impossible to read. As we add more macros, the source becomes significantly less readable, potentially until becomes only a script that's intended to be processed by hugo to generate the documentation. I want to avoid that if at all possible, and I feel strongly about it.

High quality, easily consumable docs, in both GitHub and on our website, should be a goal. If we need to have automated tooling to make sure our docs are high quality and in-sync, let's build/find that.

If we desire to stray from my goal, we should block search from indexing our GitHub repos, since it's not a useful place for someone to land when they're looking for Knative docs.

**Comment by duglin** [comment]
> Right now the process is semi-manual.

Between knative.dev being old (and people not knowing when it is old) and the docs release not being aligned with the code releases (which, BTW, I think is a huge mistake - they should all be shipped at the same time even if that means delaying the code releases), I fear it'll be hard for people to know where to look for the right stuff.

Perhaps a tangent, but while I can appreciate all of the hard work that's gone into a nicer (non-github looking web site for docs), if the pain of trying to keep it in-sync with github is costing us w.r.t. usability or time/resources, then I would prefer to just make sure our github repo docs are good and point people there, and not have the knative.dev docs at all. Much in the same way I think having two code-paths that both do the same thing, I'm not sure having two doc sites (if they can't been 100% in-sync) is a good thing either.

**Comment by samodell** [comment]
@rgregg  I'm going to set up a meeting for us to discuss further, but I do hear where you're coming from. I appreciate the sentiment of:

> If we need to have automated tooling to make sure our docs are high quality and in-sync, let's build/find that.

However, most of the tooling I'm aware of that reduces docs maintenance burden takes a hit on readability of the source (not automated site builds, but adding in variables and pulling in code, etc.).

@duglin  knative.dev is definitely a long-term investment. As Knative continues to grow, we won't be able to justify not having a professional website presence. From the beginning, we wanted to have a website to make Knative a peer to OS technologies like Istio and Kubernetes, which both have nice-looking sites.

**Comment by mattmoor** [comment]
@rgregg when this transformation first started, I definitely agreed with your sentiment.  I think this is exacerbated by the fact that this repo was all we had when we first launched the project, it is still pinned, and it still is one of our most starred repos.

However, as we start to try and do more advanced things in our documentation, it is going to become increasingly problematic to try and use pure markdown to achieve our goals.  I worry about suggestions like Richie's in the linked issue above because they may work well when first written, and make doing moderately advanced things exceptionally hard (e.g. the tab work I just did).

To @duglin 's point, I think we've gotten a bit better at shipping docs on/around release day.

Has the source of truth for docs shifted enough since this issue started that we should revisit the need for this?  Are there AIs we can take to alleviate the perception that folks should be looking here?  What is the current thinking from the various interested parties?
